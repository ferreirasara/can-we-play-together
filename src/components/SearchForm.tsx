import { LoadingButton } from "@mui/lab";
import { Alert, Box, CircularProgress, TextField, Typography } from "@mui/material";
import { useState } from "react";
import SteamidHelpModalButton from "./SteamidHelpModal";

type SearchFormProps = {
  handleSubmit: (userId1: string, userId2: string) => void
  loading: boolean
}
export default function SearchForm({
  handleSubmit,
  loading,
}: SearchFormProps) {
  const [userId1, setUserId1] = useState<string>();
  const [userId2, setUserId2] = useState<string>();
  const [error, setError] = useState<string>();
  const [errorAlertVisible, setErrorAlertVisible] = useState<boolean>(false);

  const handleButtonClick = () => {
    if (!userId1 || !userId2) {
      setErrorAlertVisible(true);
      setError('You must fill both fields.')
    } else if (userId1 === userId2) {
      setErrorAlertVisible(true);
      setError('The usernames must be different.');
    } else {
      setErrorAlertVisible(false);
      handleSubmit(userId1, userId2);
    }
  }

  return (
    <Box
      display="flex"
      flexDirection="column"
      alignItems="center"
      gap={2}
      p={1}
    >
      <Box
        display="flex"
        justifyContent="center"
        alignItems="center"
        gap={2}
      >
        <TextField
          disabled={loading}
          fullWidth
          id="user1"
          label="Your username"
          variant="outlined"
          required
          error={!!(error && !userId1)}
          onChange={(value) => setUserId1(value?.currentTarget?.value)}
        />
        <TextField
          disabled={loading}
          fullWidth
          id="user2"
          label="Your friend's username"
          variant="outlined"
          required
          error={!!(error && !userId2)}
          onChange={(value) => setUserId2(value?.currentTarget?.value)}
        />
      </Box>
      <SteamidHelpModalButton />
      {errorAlertVisible ? <Alert onClose={() => setErrorAlertVisible(false)} severity="error">{error}</Alert> : null}
      <LoadingButton
        type="submit"
        loading={loading}
        loadingIndicator={
          <Box display="flex" alignItems="center" gap={1}>
            <CircularProgress size={15} />
            <Typography>Analyzing...</Typography>
          </Box>
        }
        fullWidth
        variant="contained"
        size="large"
        onClick={handleButtonClick}
      >
        Can we play together?
      </LoadingButton>
    </Box >
  );
}
