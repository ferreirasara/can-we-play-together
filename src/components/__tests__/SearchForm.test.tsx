import renderer from 'react-test-renderer';
import SearchForm from '../SearchForm';

it('renders correctly', () => {
  const filters = renderer
    .create(<SearchForm handleSubmit={() => { }} loading={false} />)
    .toJSON();
  expect(filters).toMatchSnapshot();
});