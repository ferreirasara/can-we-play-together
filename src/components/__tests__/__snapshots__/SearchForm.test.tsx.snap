// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-ewzdx8"
>
  <div
    className="MuiBox-root css-1embff6"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="user1"
        id="user1-label"
      >
        Your username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="user1"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              Your username
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="user2"
        id="user2-label"
      >
        Your friend's username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="user2"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              Your friend's username
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <span
    className="MuiTypography-root MuiTypography-caption css-1oob06v-MuiTypography-root"
    onClick={[Function]}
  >
    How can I find my steamid?
  </span>
  <button
    className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-sppx0e-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    disabled={false}
    id=":r3:"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    Can we play together?
  </button>
</div>
`;
